content/javascript/why-amib.md

```diff
@@ -0,0 +1,28 @@
+Title: Backbone isomorphique maison : pourquoi ?
+Date: 2018-01-10
+Slug: pourquoi-backbone-isomorphique
+Author: Nicolas Bétheuil
+Tags: french,JavaScript,NodeJS,backbone
+Summary: Pourquoi-a-t'on réécrit une librairie isomorphique basée sur backbone ?
+
+Vous avez peut être remarqué, nous avons refait une partie du site il y a bien six mois maintenant : les itinéraires. Nous appelons cela le multipath (prononcer [moultipaðe](https://www.anglaisfacile.com/exercices/exercice-anglais-2/exercice-anglais-66477.php))
+
+Petit tour du propriétaire : vous pouvez dorénavant pour une recherche d'itinéraire avoir les réponses dans plusieurs modes de transport : voiture (évidemment), bus, car, taxi, VTC, transport en commun, à pied ... et beaucoup d'autres modes suivront. 
+
+Comme n'importe quelle partie du site : le [SEO](https://www.wikiwand.com/fr/Optimisation_pour_les_moteurs_de_recherche) est une exigence forte. Nos pages doivent répondre de la même manière pour un moteur de recherche que pour une navigation utilisateur classique. 
+
+Lors de la [migration de PHP vers NodeJS](http://techblog.mappy.com/mappy-com-de-php-a-nodejs.html) réalisée l'année dernière, une architecture a été posée avec une séparation nette entre la partie serveur sous `express` et la partie cliente pilotant des vues Backbone. Cela impliquait notamment deux manières complètement différentes de récupérer la donnée avant de rendre la vue. Nous souhaitions explorer d'autres pistes pour faire évoluer l'architecture sans tout refondre. La piste de l'isomorphisme a alors été suivie pour uniformiser cette récupération de donnée. Nous souhaitions également suivre de manière plus rigoureuse le design MV de backbone. 
+
+Une autre contrainte était la courbe d'apprentissage et le délai que nous avions pour faire cette évolution stratégique du produit Mappy. Nous étions une équipe de cinq développeurs, les autres connaissaient déjà [backbone](http://backbonejs.org/) et j'arrivais avec plus de connaissances sur [ReactJS](https://reactjs.org/) et la refonte d'un autre site en isomorphique. 
+
+Nous ne souhaitions pas utiliser React, son écosystème et ses minimum ~40Kb minifiés & compressés en plus de backbone pour des raisons de performance, notamment en mobile. 
```

Comment:
Je ne serais pas si catégorique. Par exemple: « Pour des raisons de "time to market", nous ne souhaitions pas  migrer tout le site vers React. Par ailleurs, l’ajouter en plus de backbone, avec son ecosystème et ses ~40Kb minifiés & compressés était problèmatique pour des raisons de performance, notamment en mobile ».